co-log/test/Property.hs

```diff
@@ -0,0 +1,59 @@
+module Main where
+
+import System.IO (BufferMode(..), hSetBuffering, stdout, stderr)
+
+import Hedgehog
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+import Colog.Pure (PureLogger, runPureLog, logMessagePure)
+import Colog.Monad (usingLoggerT, logMsg)
+import Colog.Core (LogAction)
+
+data LogAST = LogMsg
+  | AND LogAST LogAST deriving Show
+
+validateAST :: LogAST -> String -> [String]
+validateAST LogMsg msg = [msg]
+validateAST (AND a b) msg = validateAST a msg ++ validateAST b msg
```

Comment:
I feel that name is a bit misleading. This function doesn't validate AST, it's more about converting AST to the list of logger messages.

Reply:
Renamed to `listLogMessages`